### PR TITLE
feat: easy url copy for OpenAPI docs

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Implements the operation "example" column of OpenAPI docs view, adding in a code block with an easy-to-copy path URL for each operation.

## 📸 Design Screenshots

🎨 [Figma](https://www.figma.com/file/wjoC3ZVsA1kDuPsFLjEFuY/API-Template?type=design&node-id=526-33909&mode=design&t=1RqBvXcyMr4Zg23I-4)

| Before | After |
| - | - |
| before | after |
| before-createapp | after-createapp |

## 🧪 Testing

- [ ] Visit [/hcp/api-docs/vault-secrets]
    - Each operation should have a code block in the "examples" section that shows an easy-to-copy URL to the operation path
    - On larger viewports, the "examples" section should be sticky as the page is scrolled
    - On smaller viewports where the operation "examples" and "details" sections stack, the "examples" section should no longer exhibit `sticky` behaviour
    - The padding around the main area should match design specs

> **Note**: code wrapping is not yet implemented, see section below for details.

## 💭 Anything else?

Code is the URL code block is meant to wrap, but this is not currently supported in `@hashicorp/react-code-block`. It felt like there were two ways to address this:
1. Hack in some CSS to make it work.
2. Update `react-code-block` to support wrapping.

I tried `1.` first, and [it worked alright](https://dev-portal-eb0ysefgi-hashicorp.vercel.app/hcp/api-docs/vault-secrets), but felt super messy. I've got `2.` in progress at [react-components#991 - feat(code-block): add wrapCode option](https://github.com/hashicorp/react-components/pull/991), and will aim to have the better solution in place for launch.

At worst we can always drop in [the CSS override hack](https://github.com/hashicorp/dev-portal/pull/2112/commits/7ce8accdba948e2f6590fd9f469dc3692633f2fd#diff-7f25ea8f4aa909372b136066163fe34556dea1b7add9e114aa348ae28312f24d) if we urgently need to support this. 

[task]: https://app.asana.com/0/1204678746647847/1205011719572366/f
[preview]: https://dev-portal-git-zsopen-api-operation-url-hashicorp.vercel.app/
[/hcp/api-docs/vault-secrets]: https://dev-portal-git-zsopen-api-operation-url-hashicorp.vercel.app/hcp/api-docs/vault-secrets

### CODE REVIEW

**Comment by zchsh** [comment]
Note: hoping to get code block wrapping implemented in `@hashicorp/react-code-block` as part of finalizing this PR and getting the URL code block easy to use and copy.

That work is happening in [[react-components#991 - feat(code-block): add wrapCode option](https://github.com/hashicorp/react-components/pull/991)](https://github.com/hashicorp/react-components/pull/991). Will be testing canary releases off of that PR here!

File: src/views/open-api-docs-view/components/operation-examples/index.tsx [hunk 1]
```diff
@@ -3,20 +3,35 @@
...
+import CodeBlock from '@hashicorp/react-code-block'
+// Styles
+import s from './operation-examples.module.css'
 
 /**
- * TODO: implement this presentation component.
+ * Display the operation's full URL in an easy to copy code block,
+ * as a precursor to more fully built out example requests.
  */
 export function OperationExamples({
```
Comment (zchsh) [inline src/views/open-api-docs-view/components/operation-examples/index.tsx:13]: `<OperationExamples />` is really just a code block with a URL path for now.

File: src/views/open-api-docs-view/components/operation-examples/index.tsx [hunk 2]
```diff
@@ -3,20 +3,35 @@
...
-			Examples <DevCodeBlock>{JSON.stringify(operation, null, 2)}</DevCodeBlock>
-		</div>
+		<CodeBlock
+			/**
+			 * TODO: this s.codeBlock should wrap, but it's not yet supported.
+			 * `options.wrapCode` prop should be implemented in the CodeBlock
+			 * component itself instead.
+			 *
+			 * Task: https://app.asana.com/0/1204678746647847/1205233741731093/f
+			 * PR: https://github.com/hashicorp/react-components/pull/991
```
[the remark this answers is not in the document]
Comment (kendallstrautman) [reply src/views/open-api-docs-view/components/operation-examples/index.tsx:28]: Sounds good! Pragmatic 💯

File: src/views/open-api-docs-view/components/operation-examples/index.tsx (hunk 1 above)
Comment (kendallstrautman) [reply src/views/open-api-docs-view/components/operation-examples/index.tsx:13]: Do we anticipate this being more later on? If not, do we need to define it as a separate component?

**Review by kendallstrautman — APPROVED** [review]
LGTM! I think waiting to do the wrap with the react components update makes the most sense as well.

**Pull request merged**